File: ruby/raindrops/raindrops.rb
# Rain in Pl_ng
class Raindrops
  class << self
    def convert(number)
      response = mappings.keys.map do |key|
        mappings[key] if (number % key).zero?
      end.compact.join
      return number.to_s if response.empty?

      response
    end

    private

    def mappings
      {
        3 => 'Pling',
        5 => 'Plang',
        7 => 'Plong'
      }
    end
  end
end

File: go/raindrops/raindrops.go
// Package raindrops : Exercism -> Go -> Raindrops
package raindrops

import (
	"strconv"
	"strings"
)

// Convert takes an input and returns raindrops
func Convert(input int) string {
	var output []string
	if input%3 == 0 {
		output = append(output, "Pling")
	}

	if input%5 == 0 {
		output = append(output, "Plang")
	}

	if input%7 == 0 {
		output = append(output, "Plong")
	}

	if len(output) < 1 {
		return strconv.Itoa(input)
	}
	return strings.Join(output, "")
}
